modular_skyrat/modules/verbs/code/subtle.dm

```diff
@@ -83,47 +75,63 @@
 	return FALSE
 
 /datum/emote/living/subtler/run_emote(mob/user, params, type_override = null)
+	if(!can_run_emote(user))
+		user.show_message(span_warning("You can't emote at this time."))
+		return FALSE
 	var/subtler_message
 	var/subtler_emote = params
+	var/mob/target
 	if(is_banned_from(user, "emote"))
 		to_chat(user, "You cannot send subtle emotes (banned).")
 		return FALSE
 	else if(user.client && user.client.prefs.muted & MUTE_IC)
 		to_chat(user, "You cannot send IC messages (muted).")
 		return FALSE
-	else if(!params)
+	else if(!subtler_emote)
 		subtler_emote = stripped_multiline_input(user, "Choose an emote to display.", "Subtler" , null, MAX_MESSAGE_LEN)
 		if(subtler_emote && !check_invalid(user, subtler_emote))
-			var/type = input("Is this a visible or hearable emote?") as null|anything in list("Visible", "Hearable")
-			switch(type)
-				if("Visible")
-					emote_type = EMOTE_VISIBLE
-				if("Hearable")
-					emote_type = EMOTE_AUDIBLE
-				else
-					alert("Unable to use this emote, must be either hearable or visible.")
-					return
+			var/list/inview = get_hearers_in_view(1, user)
+			inview -= GLOB.dead_mob_list
+			inview.Remove(user)
+
+			for(var/mob/inviewmob in inview)
+				if(!istype(inviewmob))
+					inview.Remove(inviewmob)
+			var/list/targets = list("1-Tile Range", "Same Tile") + inview
+			target = input("Pick a target.") as null|anything in targets
+			switch(target)
+				if("1-Tile Range")
+					target = 1
+				if("Same Tile")
+					target = 0
```

Comment:
This isn't immediately clear what it's doing, it looks like it's being set to `TRUE` or `FALSE`. See if you can find a way to make it a bit clearer